src/components/cards/card-form.tsx

```diff
@@ -1,20 +1,31 @@
 import React from "react";
 import { useRecoilState } from "recoil";
-import { categoriesState, newTransactionState } from "../../data/transactions";
+import { TransactionFormI } from "../../data/transactions";
 import { Select } from "@headlessui/react";
+import { categoriesState } from "../../data/categories";
+import { DebtFormI } from "../../data/debts";
 
-export const TransactionForm = () => {
-  const [newTransaction, setNewTransaction] =
-    useRecoilState(newTransactionState);
+function isDebt(item: TransactionFormI | DebtFormI | null) {
+  return item && "beneficiary" in item;
+}
 
+export interface props<T extends DebtFormI | TransactionFormI | null> {
+  newCard: T;
+  setNewCard: (item: T) => void;
+}
+
+export const CardForm = <T extends DebtFormI | TransactionFormI | null>({
+  newCard,
+  setNewCard,
+}: props<T>) => {
   const [categories] = useRecoilState(categoriesState);
 
   const handleChange = (
     event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
   ) => {
-    if (newTransaction) {
-      setNewTransaction({
-        ...newTransaction,
+    if (newCard) {
```

Comment:
This should be more item than card. Card is just the component that we use in transaction and debts, but a transaction and debt is not the same as card
This is a card component: https://mui.com/material-ui/react-card/
Maybe you could use movement instead of card